src/storage-client/src/types/connections.rs

```diff
@@ -350,12 +321,68 @@ impl KafkaConnection {
 
         let mut context = BrokerRewritingClientContext::new(context);
         for broker in &self.brokers {
-            if let Some(aws_privatelink) = &broker.aws_privatelink {
-                context.add_broker_rewrite(
-                    &broker.address,
-                    &mz_cloud_resources::vpc_endpoint_name(aws_privatelink.connection_id),
-                    aws_privatelink.port,
-                );
+            match &broker.tunnel {
+                Tunnel::Direct => {
+                    // By default, don't override broker address lookup.
+                }
+                Tunnel::AwsPrivatelink(aws_privatelink) => {
+                    context.add_broker_rewrite(
+                        &broker.address,
+                        &mz_cloud_resources::vpc_endpoint_name(aws_privatelink.connection_id),
+                        aws_privatelink.port,
+                    );
+                }
+                Tunnel::Ssh {
+                    connection_id,
+                    connection,
+                } => {
+                    // Setup the config...
+                    let secret = connection_context
+                        .secrets_reader
+                        .read(*connection_id)
+                        .await?;
+                    let key_set = SshKeyPairSet::from_bytes(&secret)?;
+                    let key_pair = key_set.primary().clone();
+                    let ssh_tunnel_config = SshTunnelConfig {
+                        host: connection.host.clone(),
+                        port: connection.port,
+                        user: connection.user.clone(),
+                        key_pair,
+                    };
+
+                    // ...extract the broker address pieces...
+                    let mut broker_iter = broker.address.splitn(2, ':');
+
+                    let broker_host = broker_iter.next().context("BROKER is not address:port")?;
+                    let broker_port: u16 = broker_iter
+                        .next()
+                        .context("BROKER is not address:port")?
```

Comment:
Port's not required, right? And defaults to 9092?

```suggestion
                        .unwrap_or("9092")
```

Reply:
yes! i forgot we support that

Reply:
@benesch actually, i dont think consumers support that? https://jaceklaskowski.gitbooks.io/apache-kafka/content/kafka-properties-bootstrap-servers.html

Reply:
oh you added support for this in `add_broker_rewrite`...still, i think this might be new behavior!

Reply:
I don’t think so! I’m 99% sure librdkafka is fine with just `localhost`!